api/cache/object_cache.go

```diff
@@ -0,0 +1,62 @@
+package cache
+
+import (
+	"time"
+
+	"github.com/bluele/gcache"
+	"github.com/nspcc-dev/neofs-api-go/pkg/object"
+)
+
+// ObjectsCache provides interface for lru cache for objects.
+type ObjectsCache interface {
+	Get(address *object.Address) *object.Object
+	Put(address *object.Address, obj object.Object) error
+	Delete(address *object.Address) bool
+}
+
+const (
+	// DefaultObjectsCacheLifetime is a default lifetime of objects in cache.
+	DefaultObjectsCacheLifetime = time.Minute * 5
+	// DefaultObjectsCacheSize is a default maximum number of objects in cache.
+	DefaultObjectsCacheSize = 1e6
+)
+
+type (
+	// ObjectHeadersCache contains cache with objects, lifetime of cache entries and client to establish connection with NeoFS.
+	ObjectHeadersCache struct {
+		cache    gcache.Cache
+		lifetime time.Duration
+	}
+)
+
+// New creates an object of ObjectHeadersCache.
+func New(cacheSize int, lifetime time.Duration) *ObjectHeadersCache {
+	gc := gcache.New(cacheSize).LRU().Build()
+
+	return &ObjectHeadersCache{cache: gc, lifetime: lifetime}
+}
+
+// Get returns cached object.
+func (o *ObjectHeadersCache) Get(address *object.Address) *object.Object {
+	entry, err := o.cache.Get(address.String())
```

Comment:
Can we just use `address` as key?

Reply:
The gcache is implemented as a map. Address is a struct that contains cid []byte and oid []byte. If I understand correctly, we can use structs as keys and compare them if their fields are comparable, but slice, map, and function values are not comparable ([spec](https://golang.org/ref/spec#Comparison_operators)